platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookMetrics.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.address;
+
+import static com.swirlds.common.metrics.Metrics.INFO_CATEGORY;
+
+import com.swirlds.common.metrics.FunctionGauge;
+import com.swirlds.common.metrics.Metrics;
+import com.swirlds.common.system.NodeId;
+import com.swirlds.common.system.address.AddressBook;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A utility class to encapsulate the metrics for the address book.
+ */
+public final class AddressBookMetrics {
+
+    private AddressBookMetrics() {}
+
+    /**
+     * Register the metrics for the address book.
+     *
+     * @param metrics     the metrics engine
+     * @param addressBook the address book to register metrics for
+     * @param selfId      the ID of the node
+     */
+    public static void registerAddressBookMetrics(
+            @NonNull final Metrics metrics, @NonNull final AddressBook addressBook, @NonNull final NodeId selfId) {
+
+        metrics.getOrCreate(new FunctionGauge.Config<>(INFO_CATEGORY, "memberID", Long.class, selfId::getId)
```

Comment:
null checks

Reply:
All parameters are dereferenced in the constructor and will throw an exception promptly if null. Lol, and some of them did throw in unit tests that I had to fix. 🙃